sdk/highlight-next/src/edge.ts

```diff
@@ -0,0 +1,116 @@
+import type { WorkersSDK } from '@highlight-run/opentelemetry-sdk-workers'
+import {
+	H as CloudflareH,
+	HighlightEnv as CloudflareHighlightEnv,
+} from '@highlight-run/cloudflare'
+import type { ExecutionContext } from '@cloudflare/workers-types'
+import type { ResourceAttributes } from '@opentelemetry/resources/build/src/types'
+import type { NextFetchEvent, NextRequest } from 'next/server'
+
+export type HighlightEnv = {
+	projectId: string
+	otlpEndpoint?: string
+	serviceName?: string
+}
+
+type EdgeHandler = (
+	request: NextRequest,
+	event: NextFetchEvent,
+) => Promise<Response>
+
+type ExtendedExecutionContext = ExecutionContext & {
+	__waitUntilTimer?: ReturnType<typeof setInterval>
+	__waitUntilPromises?: Promise<void>[]
+	waitUntilFinished?: () => Promise<void>
+}
+
+interface HighlightInterface {
+	init: (
+		request: Request,
+		env: HighlightEnv,
+		ctx: ExtendedExecutionContext,
+		serviceName?: string,
+	) => WorkersSDK
+	consumeError: (error: Error) => void
+	sendResponse: (response: Response) => void
+	setAttributes: (attributes: ResourceAttributes) => void
+}
+
+export const H: HighlightInterface = {
+	...CloudflareH,
```

Comment:
Uses almost all of the CloudflareSDK. It just needs to patch the `init` function to polyfill.